ui/src/app/store/utils.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  createSlice,
+  PayloadAction,
+  SliceCaseReducers,
+  ValidateSliceCaseReducers,
+} from "@reduxjs/toolkit";
+
+import { TSFixMe } from "app/base/types";
+
+export interface GenericState<T> {
+  errors?: TSFixMe;
+  items?: T[];
+  loaded?: boolean;
+  loading?: boolean;
+  saved?: boolean;
+  saving?: boolean;
+  selected?: string[];
+  [key: string]: any;
+}
+
+export const createStandardSlice = <
+  T,
+  Reducers extends SliceCaseReducers<GenericState<T>>
+>({
+  name = "",
+  initialState,
+  reducers,
+}: {
+  name: string;
+  initialState: GenericState<T>;
+  reducers: ValidateSliceCaseReducers<GenericState<T>, Reducers>;
+}) =>
+  createSlice({
```

Comment:
`createSlice` is wrapped in this file to set up common state/reducers/actions. This file uses generics to refer to the model type.